Samples/WindowsCopilotRuntime/cs-wpf-sparse/WCRforWPF/MainWindow.xaml.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System.Windows;
+using System.Windows.Media.Imaging;
+using Microsoft.Graphics.Imaging;
+using Microsoft.Windows.AI.Generative;
+using Microsoft.Windows.Vision;
+using Windows.Storage;
+
+namespace WCRforWPF;
+
+/// <summary>
+/// Interaction logic for MainWindow.xaml
+/// </summary>
+public partial class MainWindow : Window
+{
+    public MainWindow()
+    {
+        InitializeComponent();
+    }
+
+    private async void SelectFile_Click(object sender, RoutedEventArgs e)
+    {
+        _currentImage = null;
+        this.Description.Text = "(picking a file)";
+        this.FileContent.Text = "(picking a file)";
+
+        var fileDialog = new Microsoft.Win32.OpenFileDialog();
+        fileDialog.DefaultExt = ".jpg";
+        fileDialog.Filter = "Image files (*.jpg, *.jpeg, *.png)|*.jpg;*.jpeg;*.png";
+        if (fileDialog.ShowDialog() == true)
+        {
+            this.FilePath.Text = fileDialog.FileName;
+            this.ProcessFile.IsEnabled = true;
+            _currentImage = await GetImageBufferFromFile(fileDialog.FileName);
+            var source = new BitmapImage(new Uri(fileDialog.FileName));
+            this.InputImage.Source = source;
+        }
+        this.Description.Text = "(done picking)";
+        this.FileContent.Text = "(done picking)";
+    }
+
+    private void PasteImageContent(object sender, RoutedEventArgs e)
+    {
+        this.Description.Text = "(pasting)";
+        this.FileContent.Text = "(pasting)";
+        var clipboard = System.Windows.Clipboard.GetDataObject();
+        if (clipboard == null)
+        {
+            this.FileContent.Text = "Can't open the clipboard, try again";
+            return;
+        }
+
+        var bitmapContent = clipboard.GetData(DataFormats.Bitmap, true) as BitmapImage;
+        if (bitmapContent == null)
+        {
+            this.FileContent.Text = "That wasn't a bitmap, try again";
+            return;
+        }
+    }
+
+    private ImageDescriptionGenerator? _ig;
+    private ImageBuffer? _currentImage;
+
+    private async Task<ImageBuffer> GetImageBufferFromFile(string path)
+    {
+        var storageFile = await StorageFile.GetFileFromPathAsync(path);
+        var inputStream = await storageFile.OpenAsync(FileAccessMode.Read);
+        var decoder = await Windows.Graphics.Imaging.BitmapDecoder.CreateAsync(inputStream);
+        var frame = await decoder.GetFrameAsync(0);
+        var softwareBitmap = await frame.GetSoftwareBitmapAsync();
+        return ImageBuffer.CreateCopyFromBitmap(softwareBitmap);
```

Comment:
This `ImageBuffer.CreateCopyFromBitmap` line crashes for me with a REGDB_E_CLASSNOTREG exception.